js/glowingbear.js

```diff
@@ -687,7 +687,26 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         }
     };
 
+    
+    $scope.parseHash = function() {
+
+        //Fill in url parameters, they take precedence over the stored settings, but store them
+        var params = $location.$$hash.split('&').map(function(val) { return {key: val.split('=')[0], value: val.split('=')[1]}; });
+        var hostParam = params.find(function(p) { return p.key === 'host'; });
+        var portParam = params.find(function(p) { return p.key === 'port'; });
+        var pathParam = params.find(function(p) { return p.key === 'path'; });
+        var passwordParam = params.find(function(p) { return p.key === 'password'; });
+        var autoconnectParam = params.find(function(p) { return p.key === 'autoconnect'; });
+        if(hostParam) { $scope.settings.host = hostParam.value; $scope.settings.hostField = hostParam.value;}
+        if(portParam) { $scope.settings.port =  parseInt(portParam.value);}
+        if(pathParam) { $scope.settings.path = pathParam.value;}
+        if(passwordParam) { $scope.settings.password = passwordParam.value;}
```

Comment:
The model for the password field is not `settings.password`, it's a password field in the scope which is saved to settings if the checkbox is set. I don't think it should just be saved to the settings if set in the hash.

Reply:
Oh ok, this is not intentional, copy pase error.